feat(dashboard_rbac): add support for related roles

amitmiran137 opened the pull request:
### SUMMARY
an alternateive for #12993
### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->

### TEST PLAN
<!--- What steps should be taken to verify the changes -->

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

nytai committed:
fix(ci): remove signature requirements for commits to master (#13034)

cdmikechen committed:
fix: untranslated text content of Dashboard page (#13024)

* fix: fix the untranslated text content of Dashboard page

* fix: add the untranslated text content of Dashboard page in messages.pot

* fix linting error

Co-authored-by: chenx <yuxiang1989>

Commit:
feat: add support for related owners

Commit:
chore: pre-commit

ktmud committed:
fix: time filter db migration optimization (#13015)

robdiciuccio committed:
fix(build): black failing on master, add to required checks (#13039)

* fix(build): black failing on master

* Add pre-commit to required checks

Commit:
fix: pre-commit

dpgaspar committed:
feat(db engines): add support for Opendistro Elasticsearch (AWS ES) (#12602)

* feat(db engines): add support for Opendistro Elasticsearch (AWS ES)

* add time grains

* lint

* bump elasticsearch-dbapi version

* add tests

* fix test

dpgaspar committed:
fix: UI toast typo (#13026)

* fix: UI toast typo

* translate

* fix error msg

Commit:
fix: pylint

villebro commented on superset/dashboards/filters.py:
```diff
@@ -138,3 +140,22 @@ def apply(self, query: Query, value: Any) -> Query:
+
+    def apply(self, query: Query, value: Optional[Any]) -> Query:
+        role_model = security_manager.role_model
+        like_value = "%" + cast(str, value) + "%"
+        return query.filter((role_model.name).ilike(like_value),)
```
instead of casting, don't we want to just return all roles if they apply? Something like
```python
if value:
    return query.filter(role_model.name.ilike(f'%{value}%'),)
return query
```
Then we could also discard the `cast`.

amitmiran137 replied:
sure 
adding tests also

betodealmeida committed:
fix: vertical scroll in query history (#13042)

betodealmeida committed:
fix: timeout context manager on Windows (#13041)

* fix: timeout decorator in Windows

* Fix lint

dbz10 committed:
style: fix typo in documentation (#13063)

* Update README.md

* Update introduction.mdx

villebro committed:
fix: sorting by saved metric (#13059)

AAfghahi committed:
fix: Disabled state button transition time (#13008)

* button fix

* tooltips disabled when it is disabled, border width changed

* added isDisabled to tooltip

* worked on transition times

* cleaned up transition to be local instead of global

* made it local

* linted

* trying to resolve a conflict

villebro committed:
fix(explore): pie chart label bugs (#13052)

simcha90 committed:
feat(style): hide dashboard header by url parameter (#12918)

* feat(native-filters): hide dashboard header bu url parameter

* lint: fix lint

* test: add tests

* test: fix test

* refactor: upgrade standalone param

* fix: pre-commit and extract to method is_standalone_mode

* test: fix tests

* test: fix tests

* fix: fix standalone statement

* refactor: fix CR notes

* chore: pre-commit

* fix: fix sticky tabs + update CR notes

* lint: fix lint

* lint: fix lint

* fix: fix CR notes

* fix: fix CR notes

* lint: fix lint

* refactor: fix cr notes

Co-authored-by: amitmiran137 <amit.miran@nielsen.com>

villebro committed:
fix(ci): multiline regex in change detection (#13075)

* fix(ci): multiline regex in change detection

* fix shellcheck issues

Commit:
fix: Retroactively add granularity param to charts (#12960)

* fix: Retroactively add granularity param to charts

* Update down revision

riahk committed:
add required * indicator to message content/notif method (#12931)

dpgaspar committed:
fix: add config to disable dataset ownership on the old api (#13051)

* fix: add config to disable dataset ownership on the old api

* fix CI docker build

* fix logic

* add deprecation comment on the config

dpgaspar committed:
fix: API tests, make them possible to run independently again (#13076)

* fix: API tests, possible to run independently

* fix workflow name

ktmud committed:
chore: lock down npm to v6 (#13069)

garden-of-delete committed:
Added a note about the ! prefix for breaking changes to CONTRIBUTING.md (#13083)

michael-s-molina committed:
Improves RTL configuration (#13079)

- Adds plugin:jest-dom/recommended to ESLint plugins list to enforce tests best practices

- Moves import @testing-library/jest-dom/extend-expect; from setup.ts to testing-library.tsx to avoid matchers collision

- Adds @testing-library/user-event to help simulate users events

kgabryje committed:
fix(explore): Enable selecting an option not included in suggestions (#13029)

simcha90 committed:
feat(cross-filters): add cross filters (#12662)

* feat: add cross filters

* refactor: fix CR notes

* lint: fix lint

* lint: fix lint

* chore: pre-commit

* refactor: under chage

* refactor: move to behaviors

* lint: fix lint

Co-authored-by: amitmiran137 <amit.miran@nielsen.com>

Commit:
Force pod restart on config changes (#13056)

simcha90 committed:
feat(native-filters): Time native filter (#12992)

* Add Time Filter component

* Improve Time Filter component

* Fix import errors

* Display Time Filter

* Remove console logs

* Change Control Panel

* Remove unnecessary files

* Use time range override

* test: fix tests

* feat: re run pipeline

* fix: fix some case for Time filter

* fix: merge with master

* use original time range

* fix height

* add cross filter behavior

* apply filters on initialization

* add applied filter to overrides

* add unit tests for merge_extra_form_data

Co-authored-by: Agata Stawarz-Pastewska <agata.stawarz-pastewska@polidea.com>
Co-authored-by: Ville Brofeldt <ville.v.brofeldt@gmail.com>

simcha90 committed:
feat(native-filters): add storybook entry for select filter (#13005)

* refactor: update native filters ignores

* refactor: use behaviour variable

* refactor: move story for Filter select

* chore: update dependencies

* lint: fix lint

* remove unnecessary changes

Co-authored-by: Ville Brofeldt <ville.v.brofeldt@gmail.com>

duynguyenhoang committed:
Update http error code from 400 to 403 (#13061)

Commit:
Custom superset_config.py + secret envs (#13096)

villebro committed:
fix(native-filters): set currentValue null when empty (#13000)

simcha90 committed:
refactor(native-filters): decouple params from filter config modal (first phase) (#13021)

* refactor: decouple params from filter config modal (first phase)

* refactor: remove unnecessary fields

* refactor: fix CR notes

* fix: fix controlValues

* refactor: fix Cr notes

Co-authored-by: amitmiran137 <amit.miran@nielsen.com>

Commit:
fix: make related roles to work with tests

Commit:
Merge branch 'master' into feat/fetch_roles_for_dashboard

* master: (30 commits)
  refactor(native-filters): decouple params from filter config modal (first phase) (#13021)
  fix(native-filters): set currentValue null when empty (#13000)
  Custom superset_config.py + secret envs (#13096)
  Update http error code from 400 to 403 (#13061)
  feat(native-filters): add storybook entry for select filter (#13005)
  feat(native-filters): Time native filter (#12992)
  Force pod restart on config changes (#13056)
  feat(cross-filters): add cross filters (#12662)
  fix(explore): Enable selecting an option not included in suggestions (#13029)
  Improves RTL configuration (#13079)
  Added a note about the ! prefix for breaking changes to CONTRIBUTING.md (#13083)
  chore: lock down npm to v6 (#13069)
  fix: API tests, make them possible to run independently again (#13076)
  fix: add config to disable dataset ownership on the old api (#13051)
  add required * indicator to message content/notif method (#12931)
  fix: Retroactively add granularity param to charts (#12960)
  fix(ci): multiline regex in change detection (#13075)
  feat(style): hide dashboard header by url parameter (#12918)
  fix(explore): pie chart label bugs (#13052)
  fix: Disabled state button transition time (#13008)
  ...

Commit:
fix: lint
